src/sonic-py-common/sonic_py_common/multi_asic.py

```diff
@@ -0,0 +1,290 @@
+
+    if asic_name.startswith(ASIC_NAME_PREFIX):
+        return asic_name[len(ASIC_NAME_PREFIX):]
+    else:
+        return None
+
+
+def get_namespaces():
+    """
+    In a multi asic platform, each ASIC is in a Linux Namespace.
+    This method returns list of all the Namespace present on the device
+
+    Note: This function can be used only when the config_db is not available
+
+    Returns:
+        List of the namespaces present in the system
+    """
+    ns_list = []
+    for path in glob.glob(NAMESPACE_PATH_GLOB):
+        ns = os.path.basename(path)
+        ns_list.append(ns)
+    return natsorted(ns_list)
+
+
+def get_all_namespaces():
+    """
+    In case of Multi-Asic platform, Each ASIC will have a linux network namespace created.
+    So we loop through the databases in different namespaces and depending on the sub_role
+    decide whether this is a front end ASIC/namespace or a back end one.
+    """
+    front_ns = []
+    back_ns = []
+    num_asics = get_num_asics()
+
+    if is_multi_asic():
+        for asic in range(num_asics):
+            namespace = "{}{}".format(ASIC_NAME_PREFIX, asic)
+            config_db = connect_config_db_for_ns(namespace)
+
+            metadata = config_db.get_table('DEVICE_METADATA')
+            if metadata['localhost']['sub_role'] == FRONTEND_ASIC_SUB_ROLE:
+                front_ns.append(namespace)
+            elif metadata['localhost']['sub_role'] == BACKEND_ASIC_SUB_ROLE:
+                back_ns.append(namespace)
+
+    return {'front_ns': front_ns, 'back_ns': back_ns}
+
+
+def get_namespace_list(namespace=None):
+    if not is_multi_asic():
+        ns_list = [DEFAULT_NAMESPACE]
+        return ns_list
+
+    if namespace is None:
+        namespaces = get_all_namespaces()
+        ns_list = namespaces['front_ns'] + namespaces['back_ns']
+    else:
+        ns_list = [namespace]
+
+    return ns_list
+
+
+def get_port_table(namespace=None):
+    """
+    Retrieves the ports from all the asic present on the devices
+
+    Returns:
+        a dict of all the ports
+    """
+    all_ports = {}
+    ns_list = get_namespace_list(namespace)
+
+    for ns in ns_list:
+        ports = get_port_table_for_asic(ns)
+        all_ports.update(ports)
+
+    return all_ports
+
+
+def get_port_table_for_asic(namespace):
+
+    config_db = connect_config_db_for_ns(namespace)
+    ports = config_db.get_table(PORT_CFG_DB_TABLE)
+    return ports
+
+
+def get_namespace_for_port(port_name):
+
+    ns_list = get_namespace_list()
+    port_namespace = None
+
+    for ns in ns_list:
+        ports = get_port_table_for_asic(ns)
+        if port_name in ports:
+            port_namespace = ns
+            break
+
+    if port_namespace is None:
+        raise ValueError('Unknown port name {}'.format(port_name))
+
+    return port_namespace
+
+
+def get_port_role(port_name, namespace=None):
+
+    ports_config = get_port_table(namespace)
+    if not ports_config[port_name].has_key('role'):
```

Comment:
we can check if port_name is present in port_config

Reply:
added a check in latest commit